src/sentry/notifications/notifications/organization_request/base.py

```diff
@@ -2,27 +2,22 @@
 
 import abc
 import logging
-from typing import TYPE_CHECKING, Any, Iterable, Mapping, MutableMapping, Sequence
+from typing import TYPE_CHECKING, Any, MutableMapping
 
-from sentry import analytics, roles
-from sentry.models import NotificationSetting, OrganizationMember, Team
+from sentry.models import Team
 from sentry.notifications.notifications.base import BaseNotification
-from sentry.notifications.notify import notification_providers
+from sentry.notifications.notifications.mixins.role_based_mixin import RoleBasedMixin
 from sentry.notifications.types import NotificationSettingTypes
-from sentry.notifications.utils.actions import MessageAction
-from sentry.types.integrations import EXTERNAL_PROVIDERS, ExternalProviders
 
 if TYPE_CHECKING:
     from sentry.models import Organization, User
 
 logger = logging.getLogger(__name__)
 
 
-class OrganizationRequestNotification(BaseNotification, abc.ABC):
+class OrganizationRequestNotification(RoleBasedMixin, BaseNotification, abc.ABC):
```

Comment:
So I had to make the `RoleBasedMixin` the left most class because it is overwriting some abstract methods on `BaseNotification`. This might be an anti-pattern, I'm not sure and I don't know what would be a better way of handling this. In a future PR, I am making a new notification that requires the role-based functionality but is NOT a subclass of `OrganizationRequestNotification`